src/Sdk/Cerbos/Sdk/CerbosClient.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Threading.Tasks;
+using Cerbos.Sdk.Response;
+
+namespace Cerbos.Sdk
+{
+    /// <summary>
+    /// CerbosClient provides a client implementation that communicates with the PDP.
+    /// </summary>
+    public sealed class CerbosClient
+    {
+        private Api.V1.Svc.CerbosService.CerbosServiceClient CerbosServiceClient { get; }
+
+        public CerbosClient(Api.V1.Svc.CerbosService.CerbosServiceClient cerbosServiceClient)
+        {
+            CerbosServiceClient = cerbosServiceClient;
+        }
+
+        /// <summary>
+        /// Send a request consisting of a principal, resource(s) & action(s) to see if the principal is authorized to do the action(s) on the resource(s).
+        /// </summary>
+        public CheckResourcesResponse CheckResources(Builder.CheckResourcesRequest request)
+        {
+            try
+            {
+                return new CheckResourcesResponse(CerbosServiceClient.CheckResources(request.ToCheckResourcesRequest()));
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Failed to check resources: ${e}");
+            }
+        }
+        
+        /// <summary>
+        /// Send an async request consisting of a principal, resource(s) & action(s) to see if the principal is authorized to do the action(s) on the resource(s).
+        /// </summary>
+        public Task<CheckResourcesResponse> CheckResourcesAsync(Builder.CheckResourcesRequest request)
```

Comment:
Doesn't this still require the `async` keyword? https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/async-return-types#tasktresult-return-type

Reply:
I was probably wrong about how C# async stuff works. From a quick glance, it looks like using `await` in the middle of things is fine. Your original code was  probably correct. Sorry.